concrete-core-fixture/src/fixture/glwe_secret_key_creation.rs

```diff
@@ -0,0 +1,98 @@
+use crate::fixture::Fixture;
+use crate::generation::synthesizing::SynthesizesGlweSecretKey;
+use crate::generation::{IntegerPrecision, Maker};
+use concrete_commons::parameters::{GlweDimension, PolynomialSize};
+use concrete_core::prelude::{GlweSecretKeyCreationEngine, GlweSecretKeyEntity};
+
+/// A fixture for the types implementing the `GlweSecretKeyCreationEngine` trait.
+pub struct GlweSecretKeyCreationFixture;
+
+#[derive(Debug)]
+pub struct GlweSecretKeyCreationParameters {
+    pub glwe_dimension: GlweDimension,
+    pub polynomial_size: PolynomialSize,
+}
+
+impl<Precision, Engine, SecretKey> Fixture<Precision, Engine, (SecretKey,)>
+    for GlweSecretKeyCreationFixture
+where
+    Precision: IntegerPrecision,
+    Engine: GlweSecretKeyCreationEngine<SecretKey>,
+    SecretKey: GlweSecretKeyEntity,
+    Maker: SynthesizesGlweSecretKey<Precision, SecretKey>,
+{
+    type Parameters = GlweSecretKeyCreationParameters;
+    type RepetitionPrototypes = ();
+    type SamplePrototypes = ();
+    type PreExecutionContext = ();
+    type PostExecutionContext = (SecretKey,);
+    type Criteria = ();
+    type Outcome = ();
+
+    fn generate_parameters_iterator() -> Box<dyn Iterator<Item = Self::Parameters>> {
+        Box::new(
+            vec![GlweSecretKeyCreationParameters {
+                glwe_dimension: GlweDimension(1),
+                polynomial_size: PolynomialSize(1024),
+            }]
+            .into_iter(),
+        )
+    }
+
+    fn generate_random_repetition_prototypes(
+        _parameters: &Self::Parameters,
+        _maker: &mut Maker,
+    ) -> Self::RepetitionPrototypes {
+    }
+
+    fn generate_random_sample_prototypes(
+        _parameters: &Self::Parameters,
+        _maker: &mut Maker,
+        _repetition_proto: &Self::RepetitionPrototypes,
+    ) -> Self::SamplePrototypes {
+    }
+
+    fn prepare_context(
+        _parameters: &Self::Parameters,
+        _maker: &mut Maker,
+        _repetition_proto: &Self::RepetitionPrototypes,
+        _sample_proto: &Self::SamplePrototypes,
+    ) -> Self::PreExecutionContext {
+    }
+
+    fn execute_engine(
+        parameters: &Self::Parameters,
+        engine: &mut Engine,
+        _context: Self::PreExecutionContext,
+    ) -> Self::PostExecutionContext {
+        let sk = unsafe {
+            engine.create_glwe_secret_key_unchecked(
+                parameters.glwe_dimension,
+                parameters.polynomial_size,
+            )
+        };
+        (sk,)
+    }
+
+    fn process_context(
+        _parameters: &Self::Parameters,
+        _maker: &mut Maker,
+        _repetition_proto: &Self::RepetitionPrototypes,
+        _sample_proto: &Self::SamplePrototypes,
+        _context: Self::PostExecutionContext,
+    ) -> Self::Outcome {
```

Comment:
Here you actually need to destroy the glwe key

Reply:
By copy-pasting the `process_context` from `lwe_ciphertext_cleartext_discarding_multiplication.rs`, I realized that the secret key is not destroyed there, is it a mistake or is it intended?

Reply:
I think it's just that in all the other tests the keys are in the repetitions so they're not destroyed when processing the context. @aPere3 correct me if I'm wrong.

Reply:
I also thought so. The thing I don't get is why we need to destroy the PostProcessingContext but not anything else.

Reply:
I think the idea is that the repetitions are always created by the core engine which is pure rust, so we don't need to call the destroy engines. However, objects in the context may have been created via other backends than core, so we must make sure to destroy them.